modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosNorth.scala

```diff
@@ -41,7 +41,12 @@ final case class GmosNorth[F[_]: Concurrent: Timer: Logger] private (
     def extractStageMode(config: CleanConfig): Either[ConfigUtilOps.ExtractFailure, GmosNorthType.StageModeNorth] =
       config.extractInstAs[NorthTypes#GmosStageMode](STAGE_MODE_PROP)
     val fpuDefault: GmosNorthType.FPUnitNorth = FPU_NONE
-    def isCustomFPU(config: CleanConfig): Boolean = extractFPU(config).map(_.isCustom()).getOrElse(false)
+    def isCustomFPU(config: CleanConfig): Boolean =
+      (extractFPU(config), extractCustomFPU(config)) match {
+        case (Right(builtIn), _) if builtIn.isCustom() => true
+        case (_, Right(_))                             => true
```

Comment:
I would have expected the previous version to work too.  How are the sequence steps generated?  There is an option that gives you the full configuration at each step and the FPU should be correct there.  If you want to send me a sequence where it is wrong, I'll take a look.

Reply:
Thanks, I looked at the sequence you sent.  It's clearly a bug in the `ocs` code but this looks like it should work around it.  :-/